sdk/tables/azure-data-tables/tests/test_encoder.py

```diff
@@ -395,21 +811,19 @@ def test_encoder_create_entity_tuples(
             }
             expected_entity = {
                 "PartitionKey": "PK2",
-                "PartitionKey@odata.type": "Edm.String",
                 "RowKey": "RK2",
-                "RowKey@odata.type": "Edm.String",
-                "Data1": 12345,
+                "Data1": "12345", # Q: Should we encode this to int type?
```

Comment:
The old encoded result is:
```json
{
  "PartitionKey": "PK2",
  "RowKey": "RK2",
  "Data1": 12345,
  "Data1@odata.type": "Edm.Int32",
  "Data2": "False",
  "Data2@odata.type": "Edm.Boolean",
  "Data4": "2022-04-01T09:30:45.000000Z",
  "Data4@odata.type": "Edm.DateTime",
  "Data5": "9b8fbe91-a1d8-46bb-99b6-09e1a6afd9cf",
  "Data5@odata.type": "Edm.Guid",
  "Data7": "3.14",
  "Data7@odata.type": "Edm.Double",
  "Data8": "9223372036854775807",
  "Data8@odata.type": "Edm.Int64"
}
```
We should re-consider the backward compatibility issue for this case.

This test will pass when Data1 value is in string, here's the uploaded data in Azure:
![image](https://github.com/Azure/azure-sdk-for-python/assets/20915761/88e14ec7-004e-41f5-8f1a-41ff02a48374)
You can see it is saved as `int`, it proves service save the data type from "odata.type" instead of the "value type".

Reply:
We've decided to pass the customer provided value in tuple for now, so this conversation is already resolved.